packages/augur-test/src/tests/state/getter/users/get-profit-loss-and-summary.test.ts

```diff
@@ -36,147 +40,502 @@ describe('State API :: Users :: ', () => {
+    categories?: string[], description?: string, outcomes?: string[], formatOutcomes?: boolean, faucet?: boolean
+  } = {}): Promise<Market> {
+  return user.createCategoricalMarket({
+    endTime: (await user.getTimestamp()).plus(timeDelta),
+    feePerCashInAttoCash: new BigNumber(marketFee),
+    affiliateFeeDivisor: new BigNumber(affiliateFee),
+    designatedReporter: designatedReporter || user.account.address,
+    extraInfo: JSON.stringify({
+      categories,
+      description,
+    }),
+    outcomes: formatOutcomes ? outcomes.map(formatBytes32String) : outcomes,
+  }, faucet);
+}
+
+async function createScalarMarket (
+  user: TestContractAPI,
+  { timeDelta = DAY,
+    marketFee = 0,
+    affiliateFee = 25,
+    designatedReporter,
+    prices = [0, 100],
+    numTicks = 100,
+    categories = ['test', 'fake', 'Categorical'],
+    description = 'test market',
+    faucet = true,
+  }: {
+    timeDelta?: number, marketFee?: number, affiliateFee?: number, designatedReporter?: string,
+    categories?: string[], description?: string, prices?: Array<number|BigNumber>, numTicks?: number, faucet?: boolean
+  } = {}): Promise<Market> {
+  return user.createScalarMarket({
+    endTime: (await user.getTimestamp()).plus(timeDelta),
+    feePerCashInAttoCash: new BigNumber(marketFee),
+    affiliateFeeDivisor: new BigNumber(affiliateFee),
+    designatedReporter: designatedReporter || user.account.address,
+    prices: prices.map((p) => new BigNumber(p)),
+    numTicks: new BigNumber(numTicks),
+    extraInfo: JSON.stringify({
+      categories,
+      description,
+    }),
+  }, faucet);
+}
+
+async function finalize(user: TestContractAPI, market: Market): Promise<void> {
+  await user.advanceTimestamp(DAY * 2);
+  await market.finalize();
+  await user.advanceTimestamp(1);
+}
+
+async function report(user: TestContractAPI, market: Market, outcome: number) {
+  const endTime = await market.getEndTime_();
+  await user.setTimestamp(endTime.plus(1));
+
+  const numTicks = await market.getNumTicks_();
+  const numOutcomes = (await market.getNumberOfOutcomes_()).toNumber();
+  const outcomeList = repeat(ZERO, numOutcomes);
+  outcomeList[outcome] = numTicks;
+  await user.doInitialReport(market, outcomeList);
+}
+
+async function reportScalar(user: TestContractAPI, market: Market, outcome: number|'invalid') {
+  const endTime = await market.getEndTime_();
+  await user.setTimestamp(endTime.plus(1));
+
+  const numTicks = await market.getNumTicks_();
+  const other = numTicks.minus(outcome);
+  const outcomeList = outcome === 'invalid' ? [numTicks, ZERO, ZERO] : [ZERO, new BigNumber(outcome), other];
+  await user.doInitialReport(market, outcomeList);
+}
+
+async function claimProceeds(market: Market, users: TestContractAPI[]) {
+  for (const user of users) {
+    await user.claimTradingProceeds(market);
+  }
+}
+
+function repeat<T>(thing: T, times: number): T[] {
```

Comment:
You should put this in augur-utils :-)

Reply:
good idea! https://github.com/AugurProject/augur/pull/7363